File: OccasionAB.cs
using System;

namespace GarageAB
{
	class OccasionAB : VoitureAB
	{
		private int nbKm;
		private string dateMiseCirculation;

		public OccasionAB(int id, string marque, string modele, int puissance, int vitesseMax, int prix, int nbKm, string dateMiseCirculation) : base(id, marque, modele, puissance, vitesseMax, prix)
		{
			this.nbKm = nbKm;
			this.dateMiseCirculation = dateMiseCirculation;
		}

		/// <summary>
		/// Get le nombre de killometres
		/// </summary>
		/// <returns></returns>
		public int GetnbKm()
		{
			return nbKm;
		}

		/// <summary>
		/// Set le nombre de kilometres
		/// </summary>
		/// <param name="value"></param>
		public void SetnbKm(int value)
		{
			nbKm = value;
		}

		public string GetdateMiseCirculation()
		{
			return dateMiseCirculation;
		}

		public void SetdateMiseCirculation(string value)
		{
			dateMiseCirculation = value;
		}

		public override void AfficherVehiculeAB()
		{
			Console.WriteLine("VoitureAB{",
			"id=", Getid(),
			", marque='", Getmarque(), '\'',
			", modele='", Getmodele(), '\'',
			", puissance=", Getpuissance(),
			", vitesseMax=", GetvitesseMax(),
			", prix=", Getprix(),
			'}'
			);
		}

	}
}

File: src/com/garageab/OccasionAB.java
package com.garageab;

public class OccasionAB extends VoitureAB {
  private int nbKM;
  private String dateMiseCirculation;

  public OccasionAB(int id, String marque, String modele, int puissance, int vitesseMax, int prix, int nbKM, String dateMiseCirculation) {
    super(id, marque, modele, puissance, vitesseMax, prix);
    this.nbKM = nbKM;
    this.dateMiseCirculation = dateMiseCirculation;
  }

  /**
   * Get le nombre de km de la voiture
   * @return le nombre de km sous forme d'entier
   */
  public int getNbKM() {
    return nbKM;
  }

  /**
   * defini le nombre de km de la voiture sous forme d'entier
   * @param nbKM
   */
  public void setNbKM(int nbKM) {
    this.nbKM = nbKM;
  }

  public String getDateMiseCirculation() {
    return dateMiseCirculation;
  }

  public void setDateMiseCirculation(String dateMiseCirculation) {
    this.dateMiseCirculation = dateMiseCirculation;
  }

  @Override
  /**
   * {@inheritDoc}
   */
  public void afficherVehiculeAB() {
    System.out.println("OccasionAB{" +
            "nbKM=" + nbKM +
            ", dateMiseCirculation='" + dateMiseCirculation + '\'' +
            ", id=" + id +
            ", marque='" + marque + '\'' +
            ", modele='" + modele + '\'' +
            ", puissance=" + puissance +
            ", vitesseMax=" + vitesseMax +
            ", prix=" + prix +
            '}');
  }
}
